### PR TITLE
Use configured "mail" attribute when fetching mail addresses from LDAP

### PR DESCRIPTION
**Pull request by photoninger**
use the attribute from config to get mail address from ldap object instead of hard coded "mail".
Signed-off-by: Bernhard Lichtinger <bernhard.lichtinger@lrz.de>

### CODE REVIEW

File: ldap_auth_provider.py [hunk 1]
```diff
@@ -216,7 +216,7 @@ async def check_auth(
                         else default_display_name
                     )
 
-                    mail = response["attributes"].get("mail", [None])
+                    mail = response["attributes"].get(self.ldap_attributes["mail"], [None])
```
Comment (clokep) [inline ldap_auth_provider.py:219]:
```suggestion
                    mail = response["attributes"].get(
                        self.ldap_attributes["mail"], [None]
                    )
```

**Review by clokep — COMMENTED** [review]
It looks like in other spots we do properly use this attribute. 👍 

Could you please [sign-off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off) on your changes?

**Comment by photoninger** [comment]
> It looks like in other spots we do properly use this attribute. 👍
> 
> Could you please [sign-off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off) on your changes?

Done.

**Review by clokep — APPROVED** [review]

**Comment by clokep** [comment]
Thanks!

**Pull request merged**